test/unit/location.d

```diff
@@ -0,0 +1,35 @@
+module location;
```

Comment:
I recommend placing this in the `lexer` subdirectory. I also recommend renaming this to something like `location_offset`. Because if you're going to add a test for each token, it's going to be quite a lot of code. I would recommend keeping only tests for the location offset in this file and then create separate files other location tests.